snowflake/src/main/java/org/apache/iceberg/snowflake/JdbcSnowflakeClient.java

```diff
@@ -298,6 +311,7 @@ public List<SnowflakeIdentifier> listIcebergTables(SnowflakeIdentifier scope) {
               conn ->
                   queryHarness.query(conn, finalQuery, TABLE_RESULT_SET_HANDLER, finalQueryParams));
     } catch (SQLException e) {
+      tryMapSnowflakeExceptionToIcebergException(SnowflakeIdentifier.Type.SCHEMA, e);
```

Comment:
Even though for now we only actually use listing at the SCHEMA layer, this method already supports listing at ROOT and DATABASE, so it's not correct to pass SCHEMA as the type here.

Why not just pass the `scope` itself into the helper method? Then the helper method can also add the identifier string to the exception message.